src/utils/functionRuntimeUtils.ts

```diff
@@ -42,33 +42,39 @@ export namespace functionRuntimeUtils {
+                        if (major === FunctionRuntimeTag.core) {
+                            message += ` ${v2}`;
+                        }
+                        const update: vscode.MessageItem = { title: 'Update' };
+                        let result: vscode.MessageItem;
 
-                        const result: vscode.MessageItem = await ext.ui.showWarningMessage(message, canUpdate ? DialogResponses.yes : DialogResponses.learnMore, DialogResponses.dontWarnAgain);
-                        if (result === DialogResponses.learnMore) {
-                            // tslint:disable-next-line:no-unsafe-any
-                            opn('https://aka.ms/azFuncOutdated');
-                        } else if (result === DialogResponses.yes) {
-                            switch (major) {
-                                case FunctionRuntimeTag.latest:
-                                    await attemptToInstallLatestFunctionRuntime('v1');
-                                case FunctionRuntimeTag.core:
-                                    await attemptToInstallLatestFunctionRuntime('v2');
-                                default:
-                                    break;
+                        do {
+                            result = canUpdate ? await ext.ui.showWarningMessage(message, update, DialogResponses.learnMore, DialogResponses.dontWarnAgain) :
+                                await ext.ui.showWarningMessage(message, DialogResponses.learnMore, DialogResponses.dontWarnAgain);
+                            if (result === DialogResponses.learnMore) {
+                                // tslint:disable-next-line:no-unsafe-any
+                                opn('https://aka.ms/azFuncOutdated');
+                            } else if (result === update) {
+                                switch (major) {
+                                    case FunctionRuntimeTag.latest:
+                                        await attemptToInstallLatestFunctionRuntime('v1');
+                                    case FunctionRuntimeTag.core:
+                                        await attemptToInstallLatestFunctionRuntime('v2');
+                                    default:
+                                        break;
+                                }
+                            } else if (result === DialogResponses.dontWarnAgain) {
+                                await updateGlobalSetting(settingKey, false);
                             }
-                        } else if (result === DialogResponses.dontWarnAgain) {
-                            await updateGlobalSetting(settingKey, false);
                         }
+                        while (result === DialogResponses.learnMore);
                     }
                 } catch (error) {
                     ext.outputChannel.appendLine(`Error occurred when checking the version of 'Azure Functions Core Tools': ${parseError(error).message}`);
```

Comment:
Just noticed this behavior:
![screen shot 2018-05-07 at 11 04 31 am](https://user-images.githubusercontent.com/11282622/39716887-751916c0-51e6-11e8-939a-a8a9304e9277.png)

Feels like an easy thing you could tack on to this PR, but I'm fine if you fix separately, too.

Reply:
Do we not want to output an error message in general or just for when the user cancels?

Reply:
I was just talking about the canceled scenario